src/nashpy/algorithms/lemke_howson_lex.py

```diff
@@ -0,0 +1,111 @@
+"""A class for the Lemke Howson algorithm with lexicographical ordering"""
+from itertools import cycle
+
+import numpy as np
+
+from nashpy.integer_pivoting import (
+    make_tableau,
+    pivot_tableau_lex,
+)
+from .lemke_howson import shift_tableau, tableau_to_strategy
+
+
+def lemke_howson_lex(A, B, initial_dropped_label=0):
+    """
+	Obtain the Nash equilibria using the Lemke Howson algorithm implemented
+	using integer pivoting.
+
+	1. Start at the artificial equilibrium (which is fully labeled)
+	2. Choose an initial label to drop and move in the polytope for which
+	   the vertex has that label to the edge
+	   that does not share that label. (This is implemented using integer
+	   pivoting)
+	3. A label will now be duplicated in the other polytope, drop it in a
+	   similar way.
+	4. Repeat steps 2 and 3 until have Nash Equilibrium.
+
+	Parameters
+	----------
+
+		initial_dropped_label: int
+
+	Returns
+	-------
+
+		equilibria: A tuple.
+
+	========
+
+		* Edited such that non-basic variables are kept track of manually and integer pivoting is done using lexicographical ordering
```

Comment:
Reword the docstring to reflect the lexicographic ordering (don't worry about the non-basic variables).

Reply:
I do not fully understand why you've modified things to keep track of the non-basic variables manually. I'm sure you have a good reason could you explain it to me please?

Reply:
I found that in the degenerate case, after pivoting, basic variables can also have columns in the tableau with only one non-zero entry. As a result, the problem with the original method of finding the non-basic variables (i.e. finding the indices of the columns with only one non-zero entry) was erroneously identifying some basic variables as non-basic. Therefore, even though after pivoting there is always one entering variable and one exiting variable, the algorithm sometimes can’t identify them because before and after pivoting they still are ‘non-basic’. 

Tracking the non-basic variables manually remedies this because it will correctly identify the entering variable (the index with only one non-zero entry

Reply:
Thanks for the explanation, it's clear now.